addons/mail/static/src/views/web/form/form_controller.js

```diff
@@ -4,8 +4,11 @@ import { patch } from "@web/core/utils/patch";
 import { FormController } from "@web/views/form/form_controller";
 
 patch(FormController.prototype, "mail/views/web", {
-    onWillLoadRoot() {
-        if (this.model.root) {
+    onWillLoadRoot(nextConfiguration) {
+        const isSameThread =
+            this.model.root?.resId === nextConfiguration.resId &&
+            this.model.root?.resModel === nextConfiguration.resModel;
+        if (isSameThread) {
```

Comment:
I tried the same change recently in the previous code and it wasn't ok. https://github.com/odoo/odoo/pull/129523

Reply:
I think its exactly what we want? If the root is reloaded due to data change, the thread will be reload because `isSameThread` will be true. On the other hand, it won't it the thread is different (and messages will be loaded from `willStart/willUpdateProps`.